packages/svelte/src/internal/client/dom/elements/attributes.js

```diff
@@ -261,20 +262,13 @@ export function set_custom_element_data(node, prop, value) {
  * @param {Record<string | symbol, any> | undefined} prev
  * @param {Record<string | symbol, any>} next New attributes - this function mutates this object
  * @param {string} [css_hash]
- * @param {boolean} [preserve_attribute_case]
- * @param {boolean} [is_custom_element]
  * @param {boolean} [skip_warning]
  * @returns {Record<string, any>}
  */
-export function set_attributes(
-	element,
-	prev,
-	next,
-	css_hash,
-	preserve_attribute_case = false,
-	is_custom_element = false,
-	skip_warning = false
-) {
+export function set_attributes(element, prev, next, css_hash, skip_warning = false) {
+	var is_custom_element = element.nodeName.includes('-');
```

Comment:
Can't we cache this in the internal `__attributes` object somehow? The namespace check is actually quite expensive it seems benchmarking.

Reply:
good idea, done